src/main/kotlin/io/curity/identityserver/plugin/username/authentication/UsernameAuthenticatorRequestHandler.kt

```diff
@@ -79,9 +81,14 @@ class UsernameAuthenticatorRequestHandler(config: UsernameAuthenticatorPluginCon
                     Response.ResponseModelScope.NOT_FAILURE)
         }
 
+        // set the template and model for responses on the NOT_FAILURE scope
+        response.setResponseModel(templateResponseModel(
+            singletonMap("username", userPreferencesManager.username as Any?),
+            templateName), Response.ResponseModelScope.NOT_FAILURE)
```

Comment:
Is the `putViewData` call above still needed with this change?

Reply:
You are right, it seems redundant with the data registered with the _NOT_FAILURE_ scope template.